### PR TITLE
Cleaning up snowflake demo code for a blog

### PR DESCRIPTION
**Pull request by sfc-gh-jreini**
# Description

Create an end to end notebook that users can try to understand the key components of the RAG.

- Create RAG in notebook with Cortex Search, Cortex Complete
- Add RAG Triad evaluation metrics
- Write experiment traces and evaluate results to Snowflake tables
- Identify low performance on evaluation metrics
- Add Context Filter Guardrails to a RAG with TruLens
- Measure performance improvement overall after adding guardrails

## Other details good to know for developers

Also update the `preload_apps` script to reflect the latest changes in the repository.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

### CODE REVIEW

File: trulens_eval/examples/snowflake_demo/feedback.py [hunk 1]
```diff
@@ -46,7 +46,7 @@ def get_feedbacks(provider_name: str, use_rag: bool = True):
     f_groundedness = (
         Feedback(
             provider.groundedness_measure_with_cot_reasons, name="Groundedness"
-        ).on(Select.RecordCalls.retrieve_context.rets[:]).on_output()
+        ).on(Select.RecordCalls.retrieve_context.rets[:].collect()).on_output()
```
Comment (sfc-gh-dkurokawa) [inline trulens_eval/examples/snowflake_demo/feedback.py:49]: what's this change doing?

File: trulens_eval/examples/snowflake_demo/feedback.py [hunk 2]
```diff
@@ -61,45 +61,16 @@ def get_feedbacks(provider_name: str, use_rag: bool = True):
...
-    f_criminality_output = Feedback(
-        provider.criminality_with_cot_reasons,
-        name="Criminality output",
-        higher_is_better=False,
-    ).on_output()
-
     if use_rag:
         return [
           f_context_relevance,
           #f_small_local_models_context_relevance,
```
Comment (sfc-gh-dkurokawa) [inline trulens_eval/examples/snowflake_demo/feedback.py:67]: remove this line too.

**Review by sfc-gh-dkurokawa — COMMENTED** [review]
I made quite a few comments in the other PR (i.e. https://github.com/Snowflake-Labs/sfquickstarts/pull/1434), that you should probably incorporate here.

File: trulens_eval/examples/snowflake_demo/preload_apps.ipynb
Comment (sfc-gh-chu) [inline trulens_eval/examples/snowflake_demo/preload_apps.ipynb:1]: may want to clear outputs as they point to specific paths on your machine

Comment (sfc-gh-chu) [inline trulens_eval/examples/snowflake_demo/preload_apps.ipynb:1]: What is `st_127_132_prompt.txt` and `st_133_136_prompt.txt`? Are these prompts I (or other users) can access somewhere?

File: trulens_eval/examples/snowflake_demo/common_ui.py [hunk 3]
```diff
@@ -1,12 +1,13 @@
 import json
 import pathlib
 import threading
-from typing import Dict, Any
+from typing import Any, Dict
 
 from conversation_manager import ConversationManager
-from feedback import get_feedbacks
-from feedback import f_context_relevance, f_small_local_models_context_relevance
 from feedback import AVAILABLE_PROVIDERS
```
Comment (sfc-gh-chu) [inline trulens_eval/examples/snowflake_demo/common_ui.py:7]: Seems like a number of format changes. Did the formatter version/config change?

**Comment by sfc-gh-chu** [comment]
Why is this merging into `snowflake_demo` and not `main`?

File: trulens_eval/examples/snowflake_demo/feedback.py [hunk 4]
```diff
@@ -24,6 +24,7 @@
 
 AVAILABLE_PROVIDERS = ["Replicate", "Cortex"]
```
Comment (sfc-gh-pdharmana) [inline trulens_eval/examples/snowflake_demo/feedback.py:26]: NIT: remove

File: trulens_eval/examples/snowflake_demo/.env.template [hunk 5]
```diff
@@ -10,3 +10,7 @@ SNOWFLAKE_SCHEMA=
 SNOWFLAKE_WAREHOUSE=
 SNOWFLAKE_ROLE=
 SNOWFLAKE_CORTEX_SEARCH_SERVICE=
+
+# Other app configurations
```
Comment (sfc-gh-pdharmana) [inline trulens_eval/examples/snowflake_demo/.env.template:14]: Why do we need this for the blog post?

**Pull request closed without merging**